src/main/java/com/emc/pravega/perf/PerfStats.java

```diff
@@ -118,44 +103,62 @@ public synchronized void printAll() {
-                count, recsPerSec, mbPerSec, totalLatency / ((double) count * 1000.0), (double) maxLatency / 1000.0,
-                percs[0] / 1000.0, percs[1] / 1000.0, percs[2] / 1000.0, percs[3] / 1000.0);
+        //double[] percs = percentiles(this.latencies, 0.5, 0.95, 0.99, 0.999);
         System.out.printf(
-                " %s FINAL:, %d, %.5f MB/sec, %.2f, %.2f, %.2f, %.2f, %.2f, %.2f\n", this.action,
-                messageSize, mbPerSec, totalLatency / ((double) count * 1000.0), (double) maxLatency / 1000.0,
-                percs[0] / 1000.0, percs[1] / 1000.0, percs[2] / 1000.0, percs[3] / 1000.0);
+                "%d records %s, %.3f records/sec, %d bytes record size, %.3f MB/sec, %.4f ms avg latency, %.4f ms max latency\n", 
+                iteration, action, recsPerSec,  messageSize, mbPerSec, totalLatency / ((double) count), (double) maxLatency);
+        /*  
+        System.out.printf("latencies percentiles:  %.4f ms 50th, %.4f ms 95th, %.4f ms 99th, %.4f ms 99.9th.\n",
+                           percs[0], percs[1], percs[2], percs[3]);
+        */
+        
     }
 
-    private long[] percentiles(long[] latencies, double... percentiles) {
-        long size = Math.min(count, latencies.length);
-        Arrays.sort(latencies, 0, (int) size);
-        long[] values = new long[percentiles.length];
+    private double[] percentiles(double[] latencies, double... percentiles) {
+        Arrays.sort(latencies, 0, (int)count);
+        double[] values = new double[percentiles.length];
         for (int i = 0; i < percentiles.length; i++) {
-            int index = (int) (percentiles[i] * size);
+            int index = (int) (percentiles[i] * count);
             values[i] = latencies[index];
         }
         return values;
     }
 
-    public CompletableFuture runAndRecordTime(Supplier<CompletableFuture> fn, long startTime, int length, Executor executor) {
-        int iter = this.iteration++;
+    public CompletableFuture runAndRecordTime(SupplierWithCE<CompletableFuture, ReinitializationRequiredException> fn, long startTime, int length) throws ReinitializationRequiredException {
         CompletableFuture  retVal = fn.get();
         if(retVal == null) {
-            record(iter, (int) (System.currentTimeMillis() - startTime) * 1000, length, System.nanoTime());
+            final long endTime = System.currentTimeMillis();
+            record(length,startTime,  endTime);
         } else {
             retVal = retVal.thenAccept((d) -> {
-                record(iter, (int) (System.currentTimeMillis() - startTime) * 1000, length, System.nanoTime());
+                final long endTime = System.currentTimeMillis();
+                record(length,startTime, endTime);
             });
         }
         return retVal;
 
     }
+
+    public CompletableFuture writeAndRecordTime(SupplierWithCE<CompletableFuture, TxnFailedException> fn, int length) throws TxnFailedException {
```

Comment:
If these two methods are going to be consistent with one another, then this one should take startTime as an argument like runAndRecordTime does.

Reply:
no, its need not be 100% consistent to each other; earlier runAndRecordTime time was used for write also, I was not comfortable  with timestamping locations in the code, hence i wrote new method:writeAndRecordTime
in the next PR i will try to merge bot the methods.